tests/test_workers/test_pkg_managers/test_gomod.py

```diff
@@ -124,6 +125,122 @@ def test_resolve_gomod(
     )
 
 
+@pytest.mark.parametrize(
+    "dep_replacement, go_list_error_pkg, expected_replace",
+    (
+        (None, "github.com/pkg/errors v1.0.0", None),
+        (
+            {"name": "github.com/pkg/errors", "type": "gomod", "version": "v1.0.0"},
+            "github.com/pkg/errors v0.9.0 github.com/pkg/errors v1.0.0",
+            "github.com/pkg/errors=github.com/pkg/errors@v1.0.0",
+        ),
+        (
+            {
+                "name": "github.com/pkg/errors",
+                "new_name": "github.com/pkg/new_errors",
+                "type": "gomod",
+                "version": "v1.0.0",
+            },
+            "github.com/pkg/errors v0.9.0 github.com/pkg/new_errors v1.0.0",
+            "github.com/pkg/errors=github.com/pkg/new_errors@v1.0.0",
+        ),
+    ),
+)
+@mock.patch("cachito.workers.pkg_managers.gomod.get_golang_version")
+@mock.patch("cachito.workers.pkg_managers.gomod.GoCacheTemporaryDirectory")
+@mock.patch("subprocess.run")
+def test_resolve_gomod_vendor_dependencies(
+    mock_run,
+    mock_temp_dir,
+    mock_golang_version,
+    dep_replacement,
+    go_list_error_pkg,
+    expected_replace,
+    tmpdir,
+    sample_deps,
+    sample_deps_replace,
+    sample_deps_replace_new_name,
+    sample_package,
+):
+    mock_cmd_output = _generate_mock_cmd_output(go_list_error_pkg)
+    # mock_gwc.return_value.cachito_strict_gomod_vendor = True
+    mock_cmd_output = _generate_mock_cmd_output(go_list_error_pkg)
+    # Mock the tempfile.TemporaryDirectory context manager
+    mock_temp_dir.return_value.__enter__.return_value = str(tmpdir)
+
+    # Mock the "subprocess.run" calls
+    run_side_effects = []
+    if dep_replacement:
```

Comment:
Could you remove the `dep_replacement` part of this test since this is well covered in other tests and your code doesn't change that feature? I like to keep tests slim and concise to make future refactoring easier.

Reply:
Alright! I was also debating about that but ended up choosing the other way...